src/main/PDF/PdfController.java

```diff
@@ -68,62 +73,115 @@ public PdfController(MongoDatabase db) {
+          responseJSON = UserMessage.USER_NOT_FOUND.toJSON();
+        } else {
+          if (check != null && req.has("otherUser")) {
+            username = check.getUsername();
+            orgName = check.getOrganization();
+            userType = check.getUserType();
+          } else {
+            username = ctx.sessionAttribute("username");
+            orgName = ctx.sessionAttribute("orgName");
+            userType = ctx.sessionAttribute("privilegeLevel");
+          }
+          // System.out.println(username + ", " + orgName + ", " + userType.toString());
+          PDFType pdfType = PDFType.createFromString(req.getString("pdfType"));
+          boolean annotated = false;
+          if (pdfType == PDFType.FORM) {
+            annotated = Objects.requireNonNull(req.getBoolean("annotated"));
+          }
+          GetAllFilesPDFService getAllFilesPDFService =
+              new GetAllFilesPDFService(
+                  db, logger, username, orgName, userType, pdfType, annotated);
+          Message response = getAllFilesPDFService.executeAndGetResponse();
+          responseJSON = response.toJSON();
+
+          if (response == PdfMessage.SUCCESS) {
+            responseJSON.put("documents", getAllFilesPDFService.getFiles());
+          }
         }
+
         ctx.result(responseJSON.toString());
       };
 
   /*
   REQUIRES 2 fields in HTTP Request
     - "pdfType": String giving PDF Type ("FORM", "APPLICATION", "IDENTIFICATION")
     - "file": the PDF file to be uploaded
+  Additional support for uploading on behalf of another user, JSON body:
+    - "otherUser": the user the file is being uploaded for
    */
   public Handler pdfUpload =
       ctx -> {
-        String username = ctx.sessionAttribute("username");
-        String organizationName = ctx.sessionAttribute("orgName");
-        UserType privilegeLevel = ctx.sessionAttribute("privilegeLevel");
-        UploadedFile file = ctx.uploadedFile("file");
-        if (file == null) {
-          ctx.result(PdfMessage.INVALID_PDF.toResponseString());
+        String username;
+        String organizationName;
+        UserType privilegeLevel;
+        JSONObject req = new JSONObject(ctx.body());
+        Message response;
+        User check = userCheck(req);
+        if (!ctx.body().isEmpty() && check == null) {
+          response = UserMessage.USER_NOT_FOUND;
         } else {
-          PDFType pdfType = PDFType.createFromString(ctx.formParam("pdfType"));
-          UploadPDFService uploadService =
-              new UploadPDFService(
-                  db,
-                  logger,
-                  username,
-                  organizationName,
-                  privilegeLevel,
-                  pdfType,
-                  file.getFilename(),
-                  file.getContentType(),
-                  file.getContent());
-          ctx.result(uploadService.executeAndGetResponse().toResponseString());
+          if (!ctx.body().isEmpty() && check != null) {
```

Comment:
Why is this method slightly different than the getDocuments one? Where is `req.has("otherUser")`?

Reply:
so getDocuments already uses the ctx.body(), so the check is a bit redundant for that method. But, in pdfIUpload, the method uses the formParams as a means to getting the pdfType. Right now, if you wanted to upload or access another users files, you just add the "otherUser" value to the BODY of the http request. I kept the two methods the same just for consistency. I can pdfUpload to have the "otherUser" in the formParam so that it's like putting in the pdfType, if that was something you would rather do. I'm also changing the key name from "otherUser" to "targetUser".

Reply:
Ok, thanks for the explanation. I think what you have now is fine if I can just add the target user in the HTTP body, but I don't really understand the backend.
Can you Slack me when you change the key so I can change the frontend code appropriately? Thanks!